test/libsolidity/ASTPropertyTest.cpp

```diff
@@ -0,0 +1,169 @@
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <liblangutil/Common.h>
+#include <libsolidity/ast/ASTJsonExporter.h>
+#include <libsolidity/interface/CompilerStack.h>
+#include <libsolutil/JSON.h>
+
+#include <test/Common.h>
+#include <test/libsolidity/ASTPropertyTest.h>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/throw_exception.hpp>
+
+#include <queue>
+
+using namespace solidity::langutil;
+using namespace solidity::frontend;
+using namespace solidity::frontend::test;
+using namespace solidity;
+using namespace std;
+
+ASTPropertyTest::ASTPropertyTest(string const& _filename):
+	TestCase(_filename)
+{
+	if (!boost::algorithm::ends_with(_filename, ".sol"))
+		BOOST_THROW_EXCEPTION(runtime_error("Invalid test contract file name: \"" + _filename + "\"."));
+
+    m_source = m_reader.source();
+    readExpectations();
+}
+
+void ASTPropertyTest::generateTestCaseValues(string& _values, bool _obtained)
+{
+    _values.clear();
+    for (auto testId: m_expectationsSequence)
+    {
+        soltestAssert(m_testCases.count(testId) > 0);
+        _values +=
+            testId +
+            ": " +
+            (_obtained ? m_testCases[testId].obtainedValue : m_testCases[testId].expectedValue)
+            + "\n";
+    }
+}
+
+vector<StringPair> ASTPropertyTest::readKeyValuePairs(string const& _input)
+{
+    vector<StringPair> result;
+    vector<string> lines;
+    boost::algorithm::split(lines, _input, boost::is_any_of("\n"));
+    for (auto const& line: lines)
+    {
+        if (line.empty())
+            continue;
+
+        vector<string> pair;
+        boost::algorithm::split(pair, line, boost::is_any_of(":"));
+        soltestAssert(pair.size() == 2);
+        for (auto& element: pair)
+        {
+            boost::trim(element);
+            soltestAssert(ranges::all_of(element, [](char c) { return isprint(c); }));
+        }
+        result.emplace_back(pair[0], pair[1]);
+    }
+    return result;
+}
+
+void ASTPropertyTest::readExpectations()
+{
+    for (auto [testId, testExpectation]: readKeyValuePairs(m_reader.simpleExpectations()))
+    {
+        m_testCases.emplace(testId, ASTPropertyTestCase{testId, "", testExpectation, ""});
+        m_expectationsSequence.push_back(testId);
+    }
+    generateTestCaseValues(m_expectation, false);
+}
+
+optional<Json::Value> ASTPropertyTest::findNode(Json::Value const& _root, string_view const& _property)
+{
+    if (!_property.empty())
+    {
+        string subNode = _property.substr(0, _property.find_first_of('.'));
+        if (subNode != _property)
+            return findNode(_root[subNode], _property.substr(_property.find_first_of('.') + 1));
```

Comment:
Oooh, OK, my bad, you do need to explicitly create a `string` here - I didn't notice that `_property` was in fact a `string_view`.